crates/shared/src/sources/balancer_v2/pool_fetching.rs

```diff
@@ -136,19 +137,47 @@ pub struct BalancerPoolFetcher {
     fetcher: Arc<dyn InternalPoolFetching>,
 }
 
+arg_enum! {
+    /// An enum containing all supported Balancer factory types.
+    #[derive(Clone, Copy, Debug)]
+    pub enum BalancerFactoryKind {
+        Weighted,
+        Weighted2Token,
+        Stable,
+    }
+}
+
+impl BalancerFactoryKind {
+    /// Returns all supported Balancer factory kinds.
+    pub fn all() -> Vec<Self> {
+        // take advantage of the auto-generated `::variants()` associated
+        // function so we don't have to keep updating this method with new kinds
+        // as they get added. A little inneficient though.
```

Comment:
Nit: typo + colloquial wording, but also perhaps we could include in the comment something about what the more efficient alternative is here and why we don't do that instead.
```suggestion
        // as they get added. Slightly inefficient.
```